cli/graphql.md

```diff
@@ -899,6 +899,100 @@ The generated resolvers would be protected like so:
+
+```javascript
+exports.handler = function (event, context) {
+  context.done(null, event.arguments.msg);
+};
+```
+
+**If you deployed your function using the 'amplify function' category**
+
+The Amplify CLI provides support for maintaining multiple environments out of the box. When you deploy a function via `amplify add function`, it will automatically add the environment suffix to your Lambda function name. For example if you create a function named **echo** using `amplify add function` in the **dev** environment, the deployed function will be named **echo-dev**. The `@function` directive allows you to use `${env}` to reference the current Amplify CLI environment.
+
+```
+type Query {
+  echo(msg: String): String @function(name: "echofunction-${dev}")
+}
+```
+
+**If you deployed your function without amplify**
+
+If you deployed your API without amplify then you must provide the full Lambda function name. If we deployed the same function with the name **echofunction** then you would have:
+
+```
+type Query {
+  echo(msg: String): String @function(name: "echofunction")
+}
+```
+
+**Structure of the AWS Lambda function event**
+
+When writing lambda function's that are connected via the `@function` directive, you can expect the following structure for the AWS Lambda event object.
+
+```javascript
+exports.handler = function (event) {
+  console.log(event);
+  // {
+  //   "typeName": "Query", /* Filled dynamically based on @function usage location */
+  //   "fieldName": "echo", /* Filled dynamically based on @function usage location */
+  //   "arguments": { /* GraphQL field arguments via $ctx.arguments */ },
+  //   "identity": { /* AppSync identity object via $ctx.identity */ },
+  //   "source": { /* The object returned by the parent resolver. E.G. if resolving field 'Post.comments', the source is the Post object. */ },
+  //   "request": { /* AppSync request object. Contains things like headers. */ },
+  //   "prev": { /* If using the built-in pipeline resolver support, this contains the object returned by the previous function. */ },
+  // }
+  context.done(null, event.arguments.msg)
+};
+```
+
+**Calling functions in different regions**
+
+By default, we expect the function to be in the same region as the amplify project. If you need to call a function in a different (or static) region, you can provide the **region** argument.
+
+```
+type Query {
+  echo(msg: String): String @function(name: "echofunction", region: "us-east-1")
+}
+```
+
+Calling functions in different AWS accounts is not supported via the @function directive but is supported by AWS AppSync.
+
+**Chaining functions**
+
+The @function directive supports AWS AppSync pipeline resolvers. That means, you can chain together multiple functions such that they are invoked in series when your field's resolver is invoked. To create a pipeline resolver that calls out to multiple AWS Lambda functions in series, use multiple `@function` directives on the field.
```

Comment:
Can you put a note about what happens under the covers? E.g. it's an AppSync Function corresponding to a Lambda function which are part of a Pipeline Resolver and called in left to right sequence based on how you use `@function` inline in your schema.

Reply:
Provided a bit more detail.

Reply:
Maybe tangential, but would there be a way to create a pipeline resolver with the default implementation as one of the steps?  Maybe I want to audit with a lambda after the normal VTL template has finished, or maybe do some related work with a lambda before allowing the default template to do it's thing.

Reply:
I would also make use of a "do your default thing, then run this lambda" or "run this lambda, then do your default thing" behavior

Reply:
@mikeparisstuff @undefobj Curious if there were any thoughts on this?

Reply:
@mwarger @davekiss I think this is a good idea and is one we have discussed before. This issue discusses handling pipeline support more generally and it would be great to get your feedback on how this can work: https://github.com/aws-amplify/amplify-cli/issues/1055.